### PR TITLE
vessel layer color select

### PR DESCRIPTION
**Pull request by tiagojsag**

### CODE REVIEW

File: app/src/components/Map/SettingsPanel.jsx [hunk 1]
```diff
@@ -1,34 +1,60 @@
...
 
   render() {
+    const colors = [
+      DEFAULT_VESSEL_COLOR,
+      '#e41ccc',
+      '#fe5f07',
+      '#a73c8a',
+      '#9e7854',
+      '#61c200',
+      '#c7ad23'
```
Comment (nerik) [inline app/src/components/Map/SettingsPanel.jsx]: Thos should be in constants.js too

File: app/src/components/Map/SettingsPanel.jsx [hunk 2]
```diff
@@ -1,34 +1,60 @@
...
+    ];
+    const colorOptions = [];
+    for (let colorIndex = 0; colorIndex <= colors.length; colorIndex += 1) {
```
Comment (nerik) [inline app/src/components/Map/SettingsPanel.jsx]: Array.map would be a more expressive option here.

File: app/src/components/Map/SettingsPanel.jsx (hunk 1 above)
Comment (tiagojsag) [inline app/src/components/Map/SettingsPanel.jsx]: these are temporary. for the final thing, we either remove the color changing option altogether, or add a proper color picker

File: app/src/components/Layers/CanvasLayer.jsx [hunk 3]
```diff
@@ -25,6 +27,15 @@ class CanvasLayer {
     }
   }
 
+  setVesselColor(vesselColor) {
+    this.vesselColor = {
+      r: parseInt(vesselColor.slice(1, 3), 16),
```
Comment (nerik) [inline app/src/components/Layers/CanvasLayer.jsx]: Why don't we store colors as r,g,b triplets (an object or a string) directly?

Comment (tiagojsag) [inline app/src/components/Layers/CanvasLayer.jsx]: you mean in the color picker itself?

**Pull request merged**